### PR TITLE
Add extra information about time synchronization needed

### PR DESCRIPTION
**Pull request by potiuk**
When you have several machines running Airflow and their time
is not synchronized, you might get very weird behaviour - some of
the log retrieval actions might randomly return "forbidden" error
because of expiring token. This is extremely difficult to
diagnose and figure out (some of our users spent days on
investigating that). Explicitly stating the requirement in the
forbidden error and whenever secret_key parameter is
mentioned, should help our users to diagnose it more easily
(and save maintainers from unnecessary questions and discussions
in Slack :))

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/config_templates/config.yml [hunk 1]
```diff
@@ -1085,6 +1085,10 @@
         Secret key used to run your flask app. It should be as random as possible. However, when running
         more than 1 instances of webserver, make sure all of them use the same ``secret_key`` otherwise
         one of them will error with "CSRF session token is missing".
+        The webserver key is also used to authorize requests to Celery workers when loga are retrieved.
```
Comment (eladkal) [inline airflow/config_templates/config.yml]:
```suggestion
        The webserver key is also used to authorize requests to Celery workers when logs are retrieved.
```

File: airflow/config_templates/default_airflow.cfg [hunk 2]
```diff
@@ -559,6 +559,10 @@ reload_on_plugin_change = False
 # Secret key used to run your flask app. It should be as random as possible. However, when running
 # more than 1 instances of webserver, make sure all of them use the same ``secret_key`` otherwise
 # one of them will error with "CSRF session token is missing".
+# The webserver key is also used to authorize requests to Celery workers when loga are retrieved.
```
Comment (eladkal) [inline airflow/config_templates/default_airflow.cfg]:
```suggestion
# The webserver key is also used to authorize requests to Celery workers when logs are retrieved.
```

File: docs/apache-airflow/configurations-ref.rst [hunk 3]
```diff
@@ -27,6 +27,11 @@ does not require all, some configurations need to be same otherwise they would n
 work as expected. A good example for that is :ref:`secret_key<config:webserver__secret_key>` which
 should be same on the Webserver and Worker to allow Webserver to fetch logs from Worker.
 
+The webserver key is also used to authorize requests to Celery workers when loga are retrieved. The token
```
Comment (eladkal) [inline docs/apache-airflow/configurations-ref.rst]:
```suggestion
The webserver key is also used to authorize requests to Celery workers when logs are retrieved. The token
```

File: docs/apache-airflow/howto/set-config.rst [hunk 4]
```diff
@@ -121,3 +121,8 @@ the example below.
     does not require all, some configurations need to be same otherwise they would not
     work as expected. A good example for that is :ref:`secret_key<config:webserver__secret_key>` which
     should be same on the Webserver and Worker to allow Webserver to fetch logs from Worker.
+
+    The webserver key is also used to authorize requests to Celery workers when loga are retrieved. The token
```
Comment (eladkal) [inline docs/apache-airflow/howto/set-config.rst]:
```suggestion
    The webserver key is also used to authorize requests to Celery workers when logs are retrieved. The token
```

File: docs/apache-airflow/upgrading-from-1-10/index.rst [hunk 5]
```diff
@@ -339,6 +339,11 @@ the only supported UI.
     this via any configuration mechanism. The 1.10.15 bridge-release modifies this feature
     to use randomly generated secret keys instead of an insecure default and may break existing
     deployments that rely on the default.
+    The webserver key is also used to authorize requests to Celery workers when loga are retrieved. The token
```
Comment (eladkal) [inline docs/apache-airflow/upgrading-from-1-10/index.rst]:
```suggestion
    The webserver key is also used to authorize requests to Celery workers when logs are retrieved. The token
```

File: docs/helm-chart/production-guide.rst [hunk 6]
```diff
@@ -111,6 +111,11 @@ Example to create a Kubernetes Secret from ``kubectl``:
 
     kubectl create secret generic my-webserver-secret --from-literal="webserver-secret-key=$(python3 -c 'import secrets; print(secrets.token_hex(16))')"
 
+The webserver key is also used to authorize requests to Celery workers when loga are retrieved. The token
```
Comment (eladkal) [inline docs/helm-chart/production-guide.rst]:
```suggestion
The webserver key is also used to authorize requests to Celery workers when logs are retrieved. The token
```

**Review by eladkal — APPROVED** [review]
We should avoid duplicate content as much as we can.
It's very hard to maintain it, if future edit will be made on one of the files we won't know we need to edit the others.

Maybe we can have the content in a single place and just link to it from all relevant files?

**Comment by potiuk** [comment]
I deliberetely put repeated content in this case. The problem with content that is "linked" from many places is that most users will not follow "see also" or "look for details". I believe some level of redundancy in "watch outs" and warnings like this is not only good but necessary - especially that (unlike code) those warnings will not get refactored or updated after they are merged.

**Comment by potiuk** [comment]
And when you see mistakes like that, it's also super easy to do search-replace if the content is the same :)

**Comment by potiuk** [comment]
Just one comment - tt's actually very similar reasoning why Tests shoudl be DAMP not DRY: https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html 

I agree that also in docs  DRY is important but when faced with the dillema DRY or IMMEDIATELY HELPFUL WITHOUT FOLLOWING MORE LINKS (when you see warnings and watchouts) I favour the other. 

BTW.  That gave me an idea - I need to come up with some catchy phrase like DAMP. I think this might be the next content of my blog post and presentation for the summit even :D.

**Comment by potiuk** [comment]
Actually - thanks @eladkal ! I was thinking for a while about a new idea for talk / blog post and your comment gave me fantastic idea. I am going to do it :)

**Comment by eladkal** [comment]
For me documentation is not different that managing DWH. You need one source of truth otherwise more often people will consume wrong/non update information but I also agree with what you said. Duplication can sometimes be OK (for example in intro/quick guide <-> full extensive guide). Ideally we should have the ability to embed text snippets so the content is written only once and the reader doesn't have to click a link nor is even aware that the content is taken from somewhere else. Something to inspire to...

> Actually - thanks @eladkal ! I was thinking for a while about a new idea for talk / blog post and your comment gave me fantastic idea. I am going to do it :)

Waiting to read it!

**Pull request merged**